process/block/metablock_request_test.go

```diff
@@ -0,0 +1,655 @@
+		headersPool.AddHeader(td[0].attestationHeaderData.headerHash, td[0].attestationHeaderData.header)
+		headersPool.AddHeader(td[1].attestationHeaderData.headerHash, td[1].attestationHeaderData.header)
+		numMissing, numAttestationMissing := mp.ComputeExistingAndRequestMissingShardHeaders(metaBlock)
+		time.Sleep(100 * time.Millisecond)
+		headersForBlock := mp.GetHdrForBlock()
+		require.Equal(t, uint32(0), numMissing)
+		require.Equal(t, uint32(0), headersForBlock.GetMissingHdrs())
+		require.Equal(t, uint32(0), numAttestationMissing)
+		require.Equal(t, uint32(0), headersForBlock.GetMissingFinalityAttestingHdrs())
+		require.Len(t, headersForBlock.GetHdrHashAndInfo(), 4)
+		require.Equal(t, uint32(0), numCallsMissingAttestation.Load())
+		require.Equal(t, uint32(0), numCallsMissingHeaders.Load())
+	})
+}
+
+func TestMetaProcessor_receivedShardHeader(t *testing.T) {
+	t.Parallel()
+	noOfShards := uint32(2)
+	td := createTestData()
+
+	t.Run("receiving the last used in block shard header", func(t *testing.T) {
+		t.Parallel()
+
+		numCalls := atomic.Uint32{}
+		arguments := createMetaProcessorArguments(t, noOfShards)
+		requestHandler, ok := arguments.ArgBaseProcessor.RequestHandler.(*testscommon.RequestHandlerStub)
+		require.True(t, ok)
+
+		requestHandler.RequestShardHeaderByNonceCalled = func(shardID uint32, nonce uint64) {
+			attestationNonce := td[shardID].attestationHeaderData.header.GetNonce()
+			if nonce != attestationNonce {
+				require.Fail(t, fmt.Sprintf("nonce should have been %d", attestationNonce))
+			}
+			numCalls.Add(1)
+		}
+
+		mp, err := blockProcess.NewMetaProcessor(*arguments)
+		require.Nil(t, err)
+		require.NotNil(t, mp)
+
+		hdrsForBlock := mp.GetHdrForBlock()
+		hdrsForBlock.SetNumMissingHdrs(1)
+		hdrsForBlock.SetNumMissingFinalityAttestingHdrs(0)
+		hdrsForBlock.SetHighestHdrNonce(0, td[0].referencedHeaderData.header.GetNonce()-1)
+		hdrsForBlock.SetHdrHashAndInfo(string(td[0].referencedHeaderData.headerHash), &blockProcess.HdrInfo{
+			UsedInBlock: true,
+			Hdr:         nil,
+		})
+
+		mp.ReceivedShardHeader(td[0].referencedHeaderData.header, td[0].referencedHeaderData.headerHash)
+
+		time.Sleep(100 * time.Millisecond)
+		require.Nil(t, err)
+		require.NotNil(t, mp)
+		require.Equal(t, uint32(1), numCalls.Load())
+		require.Equal(t, uint32(1), hdrsForBlock.GetMissingFinalityAttestingHdrs())
+	})
+
+	t.Run("shard header used in block received, not latest", func(t *testing.T) {
+		t.Parallel()
+
+		numCalls := atomic.Uint32{}
+		arguments := createMetaProcessorArguments(t, noOfShards)
+		requestHandler, ok := arguments.ArgBaseProcessor.RequestHandler.(*testscommon.RequestHandlerStub)
+		require.True(t, ok)
+
+		// for requesting attestation header
+		requestHandler.RequestShardHeaderByNonceCalled = func(shardID uint32, nonce uint64) {
+			attestationNonce := td[shardID].attestationHeaderData.header.GetNonce()
+			if nonce != attestationNonce {
```

Comment:
L270-L271 can be replaced with:
```GO
require.Equal(t, nonce, attestationNonce, fmt.Sprintf("nonce should have been %d, not %d", attestationNonce, nonce))
```

valid on all other tests

Reply:
done